component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -548,4 +571,213 @@ private int getTenantId() {
+        if (!lockAccountOnFailedAttempts) {
+            return null;
+        }
+
+        User user = getUserById(userId);
+        if (Utils.isAccountLocked(user)) {
+            return createAccountLockedResponse(userId, showFailureReason);
+        }
+
+        int maxAttempts = 0;
+        long unlockTimePropertyValue = 0;
+        double unlockTimeRatio = 1;
+
+        Property[] connectorConfigs = Utils.getAccountLockConnectorConfigs(user.getTenantDomain());
+        for (Property connectorConfig : connectorConfigs) {
+            switch (connectorConfig.getName()) {
+                case ACCOUNT_LOCKED_PROPERTY:
+                    if (!Boolean.parseBoolean(connectorConfig.getValue())) {
+                        return null;
+                    }
+                case FAILED_LOGIN_ATTEMPTS_PROPERTY:
+                    if (NumberUtils.isNumber(connectorConfig.getValue())) {
+                        maxAttempts = Integer.parseInt(connectorConfig.getValue());
+                    }
+                    break;
+                case ACCOUNT_UNLOCK_TIME_PROPERTY:
+                    if (NumberUtils.isNumber(connectorConfig.getValue())) {
+                        unlockTimePropertyValue = Integer.parseInt(connectorConfig.getValue());
+                    }
+                    break;
+                case LOGIN_FAIL_TIMEOUT_RATIO_PROPERTY:
+                    if (NumberUtils.isNumber(connectorConfig.getValue())) {
+                        double value = Double.parseDouble(connectorConfig.getValue());
+                        if (value > 0) {
+                            unlockTimeRatio = value;
+                        }
+                    }
+                    break;
+            }
+        }
+
+        Map<String, String> claimValues = getUserClaimValues(user, new String[]{
+                Constants.EMAIL_OTP_FAILED_ATTEMPTS_CLAIM, Constants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM});
+        if (claimValues == null) {
+            claimValues = new HashMap<>();
+        }
+
+        int currentAttempts = getCurrentAttempts(claimValues);
+        int failedLoginLockoutCountValue = getFailedLoginLockoutCount(claimValues);
+
+        Map<String, String> updatedClaims = new HashMap<>();
+        if ((currentAttempts + 1) >= maxAttempts) {
+            // Calculate the incremental unlock time interval in milli seconds.
+            unlockTimePropertyValue = (long) (unlockTimePropertyValue * 1000 * 60 * Math.pow(unlockTimeRatio,
+                    failedLoginLockoutCountValue));
+            populateAccountLockClaims(unlockTimePropertyValue, failedLoginLockoutCountValue, updatedClaims);
+            setUserClaimValues(user, updatedClaims);
+            FailureReasonDTO error = showFailureReason
+                    ? new FailureReasonDTO(Constants.ErrorMessage.CLIENT_ACCOUNT_LOCKED, userId)
+                    : null;
+            return new ValidationResponseDTO(userId, false, error);
+        } else {
```

Comment:
Can't we reuse the function `createAccountLockedResponse(String userId, boolean showFailureReason)`

Reply:
fixed